### ISSUE TITLE
🐛 Bug Report: `ITriggerPayload` type doesn't accept any JSON serializable object

### ISSUE THREAD

**Issue by leonardbinet** [novuhq/novu#3538]
### 📜 Description

In [documentation](https://docs.novu.co/overview/quick-start/#trigger-the-notification):
<img width="851" alt="Screenshot 2023-05-30 at 12 26 08" src="https://github.com/novuhq/novu/assets/1590371/c2e6f111-4a0a-46cd-97f2-fce7f2886ef2">
 
But the `ITriggerPayload` is actually more restrictive:
<img width="372" alt="Screenshot 2023-05-30 at 12 24 16" src="https://github.com/novuhq/novu/assets/1590371/b24c5d26-0f7d-4074-ab24-66cd2e217c17">
<img width="934" alt="Screenshot 2023-05-30 at 12 24 21" src="https://github.com/novuhq/novu/assets/1590371/67627d97-cfe3-4516-936c-709bd3b51be7">

and doesn't allow lists of objects

### 👟 Reproduction steps

```
import { ITriggerPayload } from '@novu/node'

export const myPayload: ITriggerPayload = {
  myList: [
    {
      myInnerString: 'string'
    }
  ]
}
```

and see type error in IDE

### 👍 Expected behavior

no type error in IDE

### 👎 Actual Behavior with Screenshots

<img width="934" alt="Screenshot 2023-05-30 at 12 24 21" src="https://github.com/novuhq/novu/assets/1590371/14150d30-1639-427d-90c8-ed112cb092cd">

### 📃 Provide any additional context for the Bug.

You might want to use https://github.com/microsoft/TypeScript/issues/1897#issuecomment-331765301

```
export type JSONPrimitive = string | number | boolean | null;
export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
export type JSONObject = { [member: string]: JSONValue };
export interface JSONArray extends Array<JSONValue> {}
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by p-fernandez** [comment]
Good catch! I assigned you the issue so you can submit a PR. We recently moved `ITriggerPayload` so you will have to apply the change in the Shared library (`@novu/shared`).

**Comment by scopsy** [comment]
resolved on novu/api latest sdk
